src/main/java/fibonacci/ReadFile.java

```diff
@@ -0,0 +1,38 @@
+package fibonacci;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.util.Scanner;
+import java.util.concurrent.ArrayBlockingQueue;
+import java.util.concurrent.BlockingQueue;
+
+/**
+ * Created by EddyJ on 6/28/16.
+ */
+public class ReadFile implements Runnable {
+    private BlockingQueue<Integer> queue = new ArrayBlockingQueue<Integer>(1000);
+    private File file = new File("fibdata.txt");
+
+    public ReadFile(BlockingQueue<Integer> queue) {
+        this.queue = queue;
+    }
+
+    @Override
+    public void run() {
+        Scanner scanner;
+        try {
+            scanner = new Scanner(file);
+        } catch (FileNotFoundException e) {
+            return;
+        }
+        while(scanner.hasNextInt()){
+            try {
+                queue.put(scanner.nextInt());
+            } catch (InterruptedException e) {
+                return;
```

Comment:
I would say if you have to return then you should at least print out the exception to have it available later.